src/Aquifer.Common/Services/CachingLanguageService.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Aquifer.Data;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Aquifer.Common.Services;
+
+public interface ICachingLanguageService
+{
+    Task<string?> GetLanguageCodeAsync(int languageId, CancellationToken ct);
+    Task<IReadOnlyDictionary<int, string>> GetLanguageCodeByIdMapAsync(CancellationToken ct);
+    Task<int?> GetLanguageIdAsync(string languageCode, CancellationToken ct);
+    Task<IReadOnlyDictionary<string, int>> GetLanguageIdByCodeMapAsync(CancellationToken ct);
+}
+
+public class CachingLanguageService(AquiferDbContext _dbContext, IMemoryCache _memoryCache) : ICachingLanguageService
+{
+    public async Task<string?> GetLanguageCodeAsync(int languageId, CancellationToken ct)
+    {
+        return (await GetLanguageCodeByIdMapAsync(ct))
+            .TryGetValue(languageId, out var languageCode)
+                ? languageCode
+                : null;
+    }
+
+    public async Task<IReadOnlyDictionary<int, string>> GetLanguageCodeByIdMapAsync(CancellationToken ct)
+    {
+        return (await GetDictionariesFromCacheAsync(ct))
+            .LanguageCodeByIdMap;
+    }
+
+    public async Task<int?> GetLanguageIdAsync(string languageCode, CancellationToken ct)
+    {
+        ArgumentNullException.ThrowIfNull(languageCode);
+
+        return (await GetLanguageIdByCodeMapAsync(ct))
+            .TryGetValue(languageCode, out var languageId)
+                ? languageId
+                : null;
+    }
+
+    public async Task<IReadOnlyDictionary<string, int>> GetLanguageIdByCodeMapAsync(CancellationToken ct)
+    {
+        return (await GetDictionariesFromCacheAsync(ct))
+            .LanguageIdByCodeMap;
+    }
+
+    private async Task<(IReadOnlyDictionary<int, string> LanguageCodeByIdMap, IReadOnlyDictionary<string, int> LanguageIdByCodeMap)> GetDictionariesFromCacheAsync(CancellationToken ct)
```

Comment:
Couple lines here going against the editorconfig's set length limit. Again, I'm okay with having a discussion with the team around increasing that, but at this time all of our projects have the 140 limit and I think we should stick to that for consistency.

Reply:
Yep, I wrote this before we had that discussion and missed going back to fix it.  I'll look into enforcing it somehow in Visual Studio or ideally the build itself.